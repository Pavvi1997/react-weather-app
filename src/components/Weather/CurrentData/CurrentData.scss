.current-weather {
    margin-top: 24px;
    margin-right: 140px;
    display: flex;
  }