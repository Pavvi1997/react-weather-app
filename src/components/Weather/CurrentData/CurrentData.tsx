import './CurrentData.scss'

type CurrentDataProps = {
    unit: string;
    location: string;
    weatherCity: any;
}

const CurrentData = ({ unit, location, weatherCity }: CurrentDataProps) => {

    const getFormatedDate = () => {
        const selectedDate = new Date(weatherCity?.daily[0].dt * 1000);
        var date = selectedDate.toLocaleString("en-GB", {
            day: "numeric",
            weekday: "short",
            month: "short",
        });
        var year = selectedDate.toLocaleString("en-GB", {
            year: "numeric",
        });
        var hour = selectedDate.toLocaleString("en-GB", {
            hour: "2-digit",
            minute: "2-digit",
            second: "2-digit",
        });
        return `${date} ${year} ${hour}`;
    };
    return (
        <>
            <h1>{location || ''}</h1>
            <div>{getFormatedDate()}</div>
            <div>Overcast</div>
            <div className="current-weather">
                <img
                    src={`http://openweathermap.org/img/wn/${weatherCity?.daily[0].weather[0].icon}.png`}
                    alt="" />
                <p>
                    {unit === "C"
                        ? (weatherCity.current.temp - 272.15).toFixed(0)
                        : (
                            (weatherCity.current.temp - 272.15) * (9 / 5) +
                            32
                        ).toFixed(0)}
                </p>
                ยบ{unit || ''}
            </div>
        </>
    )

}

export default CurrentData