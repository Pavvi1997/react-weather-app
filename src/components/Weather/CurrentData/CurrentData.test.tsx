import { render, screen } from "@testing-library/react";
import CurrentData from "./CurrentData";

const weatherCity = {
  current: {
    temp: 289.1,
  },
  daily: [
    {
      dt: 1661158800,
      temp: {
        min: 290.73,
        max: 293.99,
      },
      pop: 0,
      pressure: 1014,
      humidity: 72,
      rain: 0.32,
      wind_speed: 3.47,
      weather: [
        {
          icon: "04d",
        },
      ],
    },
  ],
};

describe("CurrentData", () => {
  it("Renders the CurrentData", () => {
    render(
      <CurrentData unit="C" location="london" weatherCity={weatherCity} />
    );
    expect(weatherCity).toMatchSnapshot();
    expect(screen.getByText("london", { exact: false }))
    expect(screen.getByText("Overcast", { exact: false }))
    expect(screen.findAllByText("17 ÂºC", { exact: false }))
  });
});
