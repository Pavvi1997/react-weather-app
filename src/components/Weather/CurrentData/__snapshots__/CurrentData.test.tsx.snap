// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentData Renders the CurrentData 1`] = `
Object {
  "current": Object {
    "temp": 289.1,
  },
  "daily": Array [
    Object {
      "dt": 1661158800,
      "humidity": 72,
      "pop": 0,
      "pressure": 1014,
      "rain": 0.32,
      "temp": Object {
        "max": 293.99,
        "min": 290.73,
      },
      "weather": Array [
        Object {
          "icon": "04d",
        },
      ],
      "wind_speed": 3.47,
    },
  ],
}
`;
